qiskit/visualization/timeline/events.py

```diff
@@ -54,6 +54,13 @@ def __init__(self,
         self.bit = bit
         self.instructions = instructions
 
+        # find channel duration
+        duration = 0
+        for inst in instructions:
+            if not isinstance(inst.operand, circuit.Delay):
+                duration = max(duration, inst.t0 + inst.duration)
+        self.duration = duration
```

Comment:
@itoko or @nkanazawa1989 can we add this info to the scheduled circuit itself? It would address https://github.com/Qiskit/qiskit-terra/issues/5109, and we won't have to reimplement it here.

Reply:
This is now implemented via #5182, and merged. So can we update this part of the code to use that